### PR TITLE
Remove parameters with default values from IRelationalCommand

### LINKED ISSUE #6700: Clean API of IRelationalCommand and related classes
* Methods on `IRelationalCommand` should not have parameters with default values, in particular `manageConnection` can be removed completely since assuming it's `true` always will not affect actual behavior due to how these methods are called.
 * `RelationalCommandBuilderFactory` should use a public type for the `ISensitiveDataLogger` category

### PR DESCRIPTION
**Pull request by AndriySvyryd**
Make RelationalCommandBuilderFactory use a public type for the ISensitiveDataLogger category

Fixes #6700

### CODE REVIEW

File: src/Microsoft.EntityFrameworkCore.Relational/Extensions/RelationalCommandExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+{
+    public static class RelationalCommandExtensions
+    {
+        /// <summary>
+        ///     Executes the command with no results.
+        /// </summary>
+        /// <param name="command"> The command to be executed. </param>
+        /// <param name="connection"> The connection to execute against. </param>
+        /// <returns> The number of rows affected. </returns>
+        public static int ExecuteNonQuery(
```
Comment (anpete) [inline src/Microsoft.EntityFrameworkCore.Relational/Extensions/RelationalCommandExtensions.cs]: Should these methods have Checks?

File: src/Microsoft.EntityFrameworkCore.Relational/Extensions/RelationalCommandExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+            [NotNull] this IRelationalCommand command,
+            [NotNull] IRelationalConnection connection)
+            => command.ExecuteNonQuery(connection, null);
```
Comment (anpete) [inline src/Microsoft.EntityFrameworkCore.Relational/Extensions/RelationalCommandExtensions.cs]: Add parameter name when passing null.

File: src/Microsoft.EntityFrameworkCore.Relational/Migrations/MigrationCommand.cs [hunk 3]
```diff
@@ -30,22 +30,18 @@ public class MigrationCommand
```
Comment (anpete) [inline src/Microsoft.EntityFrameworkCore.Relational/Migrations/MigrationCommand.cs]: Breaks in here?

File: src/Microsoft.EntityFrameworkCore.Relational/Storage/IRelationalCommand.cs [hunk 4]
```diff
@@ -34,83 +34,71 @@ public interface IRelationalCommand
         /// </summary>
```
Comment (anpete) [inline src/Microsoft.EntityFrameworkCore.Relational/Storage/IRelationalCommand.cs]: Breaks in here?

File: src/Microsoft.EntityFrameworkCore.Relational/Storage/Internal/RelationalCommand.cs [hunk 5]
```diff
@@ -267,21 +251,13 @@ public class RelationalCommand : IRelationalCommand
                     exception,
                     async: false);
 
-                if (openConnection && !closeConnection)
-                {
-                    connection.Close();
-                }
+                connection.Close();
```
Comment (anpete) [inline src/Microsoft.EntityFrameworkCore.Relational/Storage/Internal/RelationalCommand.cs]: Why is this changing? It seemed DRY'er in the finally and it no longer checks the flag on the error path

File: src/Microsoft.EntityFrameworkCore.Relational/Extensions/RelationalCommandExtensions.cs (hunk 1 above)
Comment (AndriySvyryd) [inline src/Microsoft.EntityFrameworkCore.Relational/Extensions/RelationalCommandExtensions.cs]: No, the delegated methods have them

File: src/Microsoft.EntityFrameworkCore.Relational/Extensions/RelationalCommandExtensions.cs (hunk 2 above)
Comment (AndriySvyryd) [inline src/Microsoft.EntityFrameworkCore.Relational/Extensions/RelationalCommandExtensions.cs]: Ok

File: src/Microsoft.EntityFrameworkCore.Relational/Migrations/MigrationCommand.cs (hunk 3 above)
Comment (AndriySvyryd) [inline src/Microsoft.EntityFrameworkCore.Relational/Migrations/MigrationCommand.cs]: It's unlikely this is used outside our code

File: src/Microsoft.EntityFrameworkCore.Relational/Storage/IRelationalCommand.cs (hunk 4 above)
Comment (AndriySvyryd) [inline src/Microsoft.EntityFrameworkCore.Relational/Storage/IRelationalCommand.cs]: It's unlikely this is used outside our code, if it is recompilation would be enough to fix.

File: src/Microsoft.EntityFrameworkCore.Relational/Storage/Internal/RelationalCommand.cs (hunk 5 above)
Comment (AndriySvyryd) [inline src/Microsoft.EntityFrameworkCore.Relational/Storage/Internal/RelationalCommand.cs]: There were two checks, now there's only one - cleaner.

Comment (anpete) [inline src/Microsoft.EntityFrameworkCore.Relational/Storage/Internal/RelationalCommand.cs]: Is there a semantic change here, though? Any user code that handles the exception will now have the connection closed always.

Comment (AndriySvyryd) [inline src/Microsoft.EntityFrameworkCore.Relational/Storage/Internal/RelationalCommand.cs]: No, finally is called before propagating the exception up the stack

**Comment by anpete** [comment]
:shipit:

**Pull request merged**